src/main/kotlin/calculator/application/Calculator.kt

```diff
@@ -0,0 +1,16 @@
+package calculator.application
+
+import calculator.domain.Expression
+
+class Calculator {
+
+    fun calculate(expression: Expression): Double {
+        expression.run {
+            var result = operands[0]
+            for (i in 1 until operands.size) {
+                result = operators[i - 1].operation(result, operands[i])
+            }
```

Comment:
```suggestion
        operators.forEachIndexed { i, operator ->
            result = operator.operation(result, operands[i + 1])
        }
```
이렇게도 나타낼 수 있어요!
- 코틀린은 내장 API로 이런것도 있어? 싶을 정도의 많은 API를 제공해요. `forEachIndexed()`도 그 중 하나입니다. 혹시 이런 것도 있나 싶을 때는 검색을 하면 보통 찾을 수 있더라구요.
- 자바는 람다 내에서 포획한 외부 변수(여기선 result)를 변경하지 못하는데요, 코틀린에서는 가능합니다. [링크](https://kotlinlang.org/docs/lambdas.html#closures)
- 이 함수형 스타일을 좀 더 활용해보면 `foldIndexed()`를 활용할 수도 있을 것 같아요 :)

Reply:
코틀린에서 람다 밖에 있는 외부 변수를 변경할 수 있는게 Closure 덕분이군요😮